src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -644,6 +649,9 @@ CreateHashDistributedTable(Oid relationId, char *distributionColumnName,
 		}
 		else
 		{
+			/* to allow concurrent run with another create_distributed_table */
+			heap_close(pgDistColocation, ExclusiveLock);
+			heap_open(DistColocationRelationId(), AccessShareLock);
 			sourceRelationId = ColocatedTableId(colocationId);
```

Comment:
It's worth adding a comment that `ColocatedTableId` takes a `ShareLock` on `pg_dist_partition` such that no tables can be dropped.